### PR TITLE
adding support for user specified loss functions

### PR DESCRIPTION
**Pull request by bjkomer**

### CODE REVIEW

**Comment by bjkomer** [comment]
An optional loss function to use can now be passed as the `loss_fn` parameter to `HyperoptEstimator`

Since the result of this function will be minimized, if you are using a metric where greater values are better, you can wrap it in a function that inverts the result. An example using F1 Score:

```
from sklearn.metrics import f1_score

def f1_loss(target, pred):
	return -f1_score(target, pred)

estim = HyperoptEstimator(loss_fn=f1_loss)
```

Fixes #46
